File: apps/zxtune-android/zxtune/src/main/java/app/zxtune/fs/StubObject.java
/**
 * @file
 * @brief Stub object base for safe inheritance
 * @author vitamin.caig@gmail.com
 */

package app.zxtune.fs;

import android.support.annotation.Nullable;

abstract class StubObject implements VfsObject {

  @Override
  public String getDescription() {
    return "";
  }

  @Override
  @Nullable
  public Object getExtension(String id) {
    return null;
  }
}

File: apps/zxtune-android/zxtune/src/main/java/app/zxtune/fs/StubObject.kt
/**
 * @file
 * @brief Stub object base for safe inheritance
 * @author vitamin.caig@gmail.com
 */
package app.zxtune.fs

// TODO: remove
abstract class StubObject : VfsObject {
    override val description
        get() = ""

    override fun getExtension(id: String): Any? = null
}
